### PR TITLE
Only disable `parallel` on workqueue backend

### PR DESCRIPTION
**Pull request by max-sixty**
Closes https://github.com/numbagg/numbagg/issues/209

### CODE REVIEW

File: numbagg/decorators.py [hunk 1]
```diff
@@ -596,7 +601,23 @@ def _create_gufunc(self, target):
...
 
 
-def _is_in_thread_pool():
+def _is_in_unsafe_thread_pool() -> bool:
     current_thread = threading.current_thread()
     # ThreadPoolExecutor threads typically have names like 'ThreadPoolExecutor-0_1'
-    return "ThreadPoolExecutor" in current_thread.name
+    return current_thread.name.startswith(
+        "ThreadPoolExecutor"
+    ) and _thread_backend() in ["workqueue" or None]
```
Comment (flying-sheep) [inline numbagg/decorators.py:609]: `["workqueue" or None]` == `["workqueue"]`

Comment (max-sixty) [reply numbagg/decorators.py:609]: > `["workqueue" or None]` == `["workqueue"]`

WDYM?

To the extent the proposal is just list -> set — fine , would take a PR...

Comment (dcherian) [reply numbagg/decorators.py:609]: I think he means `_thread_backend() in ["workqueue", None]` the current one doesn't check `None`

Comment (max-sixty) [reply numbagg/decorators.py:609]: Oh!! Gosh, can't believe I missed that, and on the second look too. Fixed in #333

**Pull request merged**